### DISCUSSION TITLE
[Game] Alltalk Plugin for Final Fantasy XIV

### DISCUSSION THREAD

**Discussion by RenNagasaki** [erew123/alltalk_tts#441] [General]
Hey,
I'm a longtime player of FFXIV and somehow got into Dalamud( the main entry for any modding/plugin for FFXIV).
I missed any option for fully voiced story gameplay since the only option available was using elevenlabs or local system tts.
That's why I took it on me to create a full implementation of alltalk tts into the game.

The plugin can connect to your local or remote hosted instance of alltalk and utilize it to fully voice all dialogues/chats and even random npc bubbles ingame.

https://github.com/RenNagasaki/Echokraut

I created another project which prepares training data for alltalk from the FFXIV Game Data. It exports and converts the already existing story voicing to create a dataset. So you can train on original voices for the perfect experience.

https://github.com/RenNagasaki/Echokraut-Tools

If you have any questions, feel free to reach out to me.

@erew123  I saw the Obsidian Plugin post and thought I could post mine aswell. If that's not okay with you, I can take this post down. 😁

**Comment by erew123** [comment]
@RenNagasaki 

Hey sure happy to! :) Does this work for you? https://github.com/erew123/alltalk_tts/wiki/Final-Fantasy-XIV-(FFXIV)

Can change it if you want.

As an aside, I know you use Streaming support mainly. in between dealing with life stuff, Ive managed to get a bit of time to do things recently (though I have to go away again today now, so *shrug*), I will be adding Openvoice as a pipeline in my next release. Its almost done, but time is not on my side today.

So there will be both RVC (which now has caching built in for speed) and Openvoice, which should run pretty damn fast! Both are very low VRAM requirements. RVC obviously needs voice models (Training for RVC voice models will be in my next release), Openvoice just uses wav files, same as XTTS......

So, in theory,.... in theory..... (I added the second just to push my belief), you should be able to use something like Piper TTS for ultra fast, low RAM TTS generation and then either RVC (which may be slower and needs models training) OR Openvoice which should be speedy, to make the TTS sound like the WAV sample voice you give it.... and as I say... it **SHOULD** be fast and maybe tops 600MB or VRAM (the model it loads into ram is only 128MB on disk)

So you may be interested in giving that a shot as it may be a nice alternative TTS option.

Anyway, gotta run, need to go finish packing etc.

**Reply by RenNagasaki** [reply]
Awesome! Thank you!
Just a small fix.
![grafik](https://github.com/user-attachments/assets/2ca06fa5-4ac5-47e8-a050-37faf904bf83)
Tools don't need Dalamud. It's a small standalone software.

**Reply by RenNagasaki** [reply]
Oh, this sentence also should only mention one plugin. 😄 
![grafik](https://github.com/user-attachments/assets/35927ce6-0749-46c3-81f4-a460f584e4dd)

**Reply by erew123** [reply]
done.
